packages/aws-cdk/lib/archive.ts

```diff
@@ -4,6 +4,18 @@ import fs = require('fs-extra');
 import glob = require('glob');
 import path = require('path');
 
+const appendFiles = async (directory: string, files: string[], archive: archiver.Archiver): Promise<void> => {
+  await Promise.all(files.map(async file => {
```

Comment:
This risk randomizing the order of objects in the ZIP stream, I would father process strictly sequentially (`await` in a `for` loop).